PWGHF/TableProducer/trackToCollisionAssociator.cxx

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file trackToCollisionAssociator.cxx
+/// \brief Associates tracks to collisions considering ambiguities
+///
+/// \author
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/runDataProcessing.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod;
+
+struct HfTrackToCollisionAssociation {
+  Produces<HfTrackAssoc> association;
+  Produces<HfTrackCompColls> reverseIndices;
+
+  Configurable<float> nSigmaForTimeCompat{"nSigmaForTimeCompat", 3.f, "number of sigmas for time compatibility"};
+
+  using TracksWithSel = soa::Join<Tracks, TracksExtra, TrackSelection>;
+  Filter trackFilter = requireGlobalTrackWoDCAInFilter();
+
+  void processAssocWithTime(Collisions const& collisions,
+                            TracksWithSel const& tracksUnfiltered,
+                            soa::Filtered<TracksWithSel> const& tracks,
+                            BCs const& bcs)
+  {
+    std::vector<int>** collsPerTrack = new std::vector<int>*[tracksUnfiltered.size()];
```

Comment:
It would be better to just nest vectors: `std::vector<std::vector<int>>`, `new` should be avoided.

Reply:
But we only have it sparsely filled. Why avoid new?

Reply:
The vector of vectors can have empty vectors as entries, there is a negligible overhead. It is better to avoid direct memory allocations simply because it would protect from possible memory issues.

Reply:
Yes, I understand that we can have empty vectors but we need to allocate all of them and here we process the tracks table. Given that we have a large fraction empty (we only process tracks which are accepted by the track selection), I think the approach with allocation is better. It is a service task and we can make sure we don't produce leaks I think :)

Reply:
Then one can do `std::vector<std::unique_ptr<std::vector<int>>>`, it would be still be sparse but the allocations are better managed.

Reply:
I have implemented it and I directly initialised it to the number of tracks in the definition (I checked and indeed it initialises each element to `nullptr`)